ctapipe/reco/hillas_intersection.py

```diff
@@ -0,0 +1,405 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+
+TODO:
+ - Speed tests, need to be certain the looping on all telescopes is not killing 
+performance
+- Introduce new weighting schemes
+- Make intersect_lines code more readable
+
+"""
+import numpy as np
+import itertools
+import astropy.units as u
+from ctapipe.reco.reco_algorithms import Reconstructor
+from ctapipe.io.containers import ReconstructedShowerContainer
+from ctapipe.coordinates import *
+from ctapipe.instrument import get_atmosphere_profile_functions
+
+__all__ = [
+    'HillasIntersection'
+]
+
+
+class HillasIntersection(Reconstructor):
+    '''
+    This class is a simple re-implementation of Hillas parameter based event 
+    reconstruction.
+    e.g. https://arxiv.org/abs/astro-ph/0607333
+
+    In this case the Hillas parameters are all constructed in the shared angular (
+    Nominal) system. Direction reconstruction is performed by extrapolation of the 
+    major axes of the Hillas parameters in the nominal system and the weighted average 
+    of the crossing points is taken. Core reconstruction is performed by performing the 
+    same procedure in the tilted ground system.
+
+    The height of maximum is reconstructed by the projection os the image centroid onto 
+    the shower axis, taking the weighted average of all images.
+
+    Uncertainties on the positions are provided by taking the spread of the crossing 
+    points, however this means that no uncertainty can be provided for multiplicity 2 
+    events.
+
+    '''
+
+    def __init__(self, atmosphere_profile_name="paranal"):
+
+        # We need a conversion function from height above ground to depth of maximum
+        # To do this we need the conversion table from CORSIKA
+        self.thickness_profile, self.altitude_profile = get_atmosphere_profile_functions(
+            atmosphere_profile_name)
+
+    def predict(self, hillas_parameters, tel_x, tel_y, array_direction):
```

Comment:
That would be good to have the same interface.  The `inst` object is being phased out (see #456  that I will merge in today, I think, unless there are major issues), but it will still contain the necessary info, but wrapped in a SubarrayDescription object that has the telescope positions, camera geometries, optics descriptions, etc, already constructed (so no more need for `.guess` methods.  Therefore I'd suggest having a `SubarrayDescription` as an input to the constructor of a `Reconstructor`, then the `inst` object is no longer needed. 

e.g. 
```python
subarray = ... # a SubArrayDescription object
reco = HillasReconstructor(subarray)
```